sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -478,4 +479,33 @@ class DataTypeSuite extends SparkFunSuite {
 
     assert(result === expected)
   }
+
+  test("Allow UserDefinedType to be merged into a native DateType") {
+    // In this testcase we have stored the XMLGregorianCalendar as a Timestamp
+    // And we want to read this later on as a regular TimestampType
+    val left = StructType(
+      StructField("a", TimestampType) :: Nil)
+
+    // We should be able to merge these types since the
+    // CustomXMLGregorianCalendarType maps to a TimestampTypes
+    val right = StructType(
+      StructField("a", new MyXMLGregorianCalendarUDT) :: Nil)
+
+    // We expect the DataType to be returned instead of the UserDefinedType
+    assert(left.merge(right) === left)
+  }
+
+  test("Disallow DataType to be merged into UserDefinedType") {
+    val right = StructType(
+      StructField("a", TimestampType) :: Nil)
```

Comment:
nit: You don't need a new line: `val right = StructType(StructField("a", TimestampType) :: Nil)`

Reply:
Sure